### PR TITLE
Feature/dump manager

### PR DESCRIPTION
**Pull request by lambdart**
Hey, Julio

This is the first version of the Dumpable and DumpManager classes for discussion.

Cya :)

### CODE REVIEW

File: src/utils/dump.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Abstraction of the dumpable object.
+ */
+class Dumpable {
+public:
+  /**
+   * Pure virtual function to be implemented.
+   * The function should dump implemented by the methods that are dumpable.
+   */
+  virtual void dump() = 0;
```
Comment (jcarraror) [inline src/utils/dump.h:22]: I think `virtual Bytes dump() const = 0;` would be more suitable.

File: src/utils/dump.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+/**
+ * Dumpable management.
+ * Used to store dumpable objects in memory.
+ */
+class DumpManager {
+private:
+  std::vector<Dumpable*> dumpable_;
```
Comment (jcarraror) [inline src/utils/dump.h:31]: Why not `std::vector<std::reference_wrapper<Dumpable>>`? I think that Dumpable cannot be null in this context.

**Review by jcarraror — CHANGES_REQUESTED** [review]

File: src/utils/dump.h (hunk 2 above)
Comment (lambdart) [reply src/utils/dump.h:31]: ok

File: src/utils/dump.h (hunk 1 above)
Comment (lambdart) [reply src/utils/dump.h:22]: ok

**Pull request merged**